cmd/status/printers/table_printer.go

```diff
@@ -0,0 +1,385 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package printers
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/utils/integer"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/polling/collector"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/polling/event"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+)
+
+const (
+	// updateInterval defines how often the printer will update the UI.
+	updateInterval = 1 * time.Second
+)
+
+// printContentFunc defines the function type used by the printer to ask a ColumnDef
+// to output the appropriate content for a given resource and column.
+type printContentFunc func(w io.Writer, width int, resource *event.ResourceStatus) (int, error)
+
+// ColumnDef defines the properties of every supported column by the printer.
+type ColumnDef struct {
+	// name of the column. This is just a string that is unique among all the defined columns
+	name string
+	// header defines the title of the column that will be printed in the table.
+	header string
+	// width defines the width of the column
+	width int
+	// printContent defines a function that will be called by the printer
+	// to output information into the given cell in the table for a provided
+	// ResourceStatus. This function is responsible for trimming any content that
+	// is too wide and takes care of setting the color of anything that is printed.
+	printContent printContentFunc
+}
+
+var (
+	// columns contains a list of ColumnDefs, one for each of the columns that
+	// is supported by the printer.
+	columns = []ColumnDef{
+		{
+			// Column containing the namespace of the resource.
+			name:   "namespace",
+			header: "NAMESPACE",
+			width:  10,
+			printContent: func(w io.Writer, availableWidth int, resource *event.ResourceStatus) (int, error) {
```

Comment:
a/availableWidth/w/

vars with tiny scopes should have tiny names.  We know this is width from the type def.

Reply:
ditto below

Reply:
Renamed this variable to `width`. `w` is already used by the `io.Writer`